librad/src/git/storage/pool.rs

```diff
@@ -57,8 +57,14 @@ impl AsMut<Storage> for PooledRef {
     }
 }
 
-impl From<Object<Storage, Error>> for PooledRef {
-    fn from(obj: Object<Storage, Error>) -> Self {
```

Comment:
Hm wait, so you need the secret key to create a pool, but you can downgrade the borrowed object to read only.

In #693 I ran into the situation that the storage has to be initialised eagerly, so the locking gymnastics for the lazy unit could also go away entirely (if we have some witness you need to bring to the pool). Then, we could have either ro or rw pools.

Reply:
I'm not actually quite sure what you mean here :) This initial iteration was just to get things working.

I think the simplest solution is to have a ro pool and keep this pool as the rw pool. Do you see a better way of doing it?

Reply:
So I added the witness construction but held off on modifying `net::Peer` any more than adding a function: `using_read_only`, which is just an easy way to say one is using `ReadOnly` rather than `Storage`